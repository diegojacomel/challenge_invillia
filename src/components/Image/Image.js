/** Modules */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const FigureStyle = styled.figure`
    display: inline-block;
    margin: ${props => props.margin ? props.margin : "0"};
    width: ${props => props.size ? props.size : "100%"};
    max-width: ${props => props.theme.spacing.full};
`

const Figure = ({ children, size, margin }) => (
    <FigureStyle size={size} margin={margin}>
        {children}
    </FigureStyle>
)

const ImgStyle = styled.img`
    display: block;
    max-width: ${props => props.theme.spacing.full};
`

const Img = ({ src, alt, title }) => (
    <ImgStyle src={src} alt={alt} title={title} />
)

function Image({ src, alt, title, size, margin }) {
    return (
        <Figure id="testFigure" size={size} margin={margin}>
            <Img src={src} alt={alt} title={title} id="testImg" />
        </Figure>
    )
}

Image.propTypes = {
    src: PropTypes.string,
    alt: PropTypes.string,
    title: PropTypes.string,
    size: PropTypes.string,
    margin: PropTypes.string
}

Image.defaultProps = {
    src: "#",
    alt: "Image",
    title: "Image",
    size: "100%",
    margin: "0"
}

export default Image;
