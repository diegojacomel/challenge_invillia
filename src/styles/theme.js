const primaryTheme = {
    colors: {
        primary: '#007bff',
        success: '#28a745',
        secondary: '#6c757d',
        warning: "#ffc107",
        black: '#000000',
        white: '#FFFFFF',
        gray: '#cccccc',
        grayLight: '#F2F2F2',
        grayDark: '#333333'
    },
    fontSize: {
        xs: "14px",
        sm: "16px",
        md: "20px",
        lg: "25px",
        hg: "40px",
        hhg: "60px"
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '80px',
        ggt: '100px',
        hg: '200px',
        hhg: '400px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    },
    doubleSpacing: {
        xs: "2px 5px",
        sm: "3px 10px",
        md: "5px 12px",
        lg: "8px 16px",
        xg: "10px 20px",
        hg: "20px 40px"
    },
    rounded: {
        none: '0',
        xs: '2px',
        sm: '3px',
        md: '5px',
        lg: '10px'
    }
}

export {
    primaryTheme
}