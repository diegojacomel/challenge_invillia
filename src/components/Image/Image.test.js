// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/theme';

// Component
import Image from './Image';

describe('Image', () => {
    const wrapper = shallow(<Image />);

    test('Render without crashing', () => {
        shallow(<Image />);
    })

    test('It exist', () => {
        expect(wrapper.find('#testFigure')).toExist();
        expect(wrapper.find('#testImg')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testFigure');
        expect(wrapper).toContainMatchingElement('#testImg');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testFigure')).toHaveDisplayName('Figure');
        expect(wrapper.find('#testImg')).toHaveDisplayName('Img');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testFigure')).toMatchSelector('Figure');
        expect(wrapper.find('#testImg')).toMatchSelector('Img');
    })

    test('Have props', () => {
        expect(wrapper.find('#testImg')).toHaveProp('title');
        expect(wrapper.find('#testImg')).toHaveProp('src');
        expect(wrapper.find('#testImg')).toHaveProp('alt');
        expect(wrapper.find('#testFigure')).toHaveProp('size');
        expect(wrapper.find('#testFigure')).toHaveProp('margin');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Image src="#" alt="alt" title="title" />
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})