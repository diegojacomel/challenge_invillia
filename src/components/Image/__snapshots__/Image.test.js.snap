// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Render correctly 1`] = `
<figure
  className="sc-bdVaJa jbejMB"
  size="100%"
>
  <img
    alt="alt"
    className="sc-bwzfXH hgonml"
    src="#"
    title="title"
  />
</figure>
`;
